staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -209,38 +209,35 @@ func (o *DebugOptions) Validate(cmd *cobra.Command) error {
 
 // Run executes a kubectl debug.
 func (o *DebugOptions) Run(f cmdutil.Factory, cmd *cobra.Command) error {
+	ctx := context.Background()
+
 	r := o.builder.
 		WithScheme(scheme.Scheme, scheme.Scheme.PrioritizedVersionsAllGroups()...).
-		NamespaceParam(o.Namespace).DefaultNamespace().ResourceNames("pods", o.PodNames...).
+		NamespaceParam(o.Namespace).DefaultNamespace().ResourceNames("pods", o.TargetNames...).
```

Comment:
With this in, you need to update the examples to use other resources, but that can be done in the next PR.